cmd/root.go

```diff
@@ -50,8 +54,22 @@ HPC deployments on the Google Cloud Platform.`,
 // Execute the root command
 func Execute() error {
 	if len(GitCommitInfo) > 0 {
+		dir, err := os.Getwd()
+		if err == nil {
+			repo, err := git.PlainOpen(dir)
+			if err == nil {
+				head, err := repo.Head()
+				if err == nil && GitCommitHash != head.Hash().String() {
+					b := head.Name().Short()
+					h := head.Hash().String()
+					fmt.Fprintf(os.Stderr, "WARNING: ghpc binary was built from a different commit (%s/%s) than the working directory (%s/%s). You can rebuild the binary by running 'make'\n",
+						GitBranch, GitCommitHash[0:7], b, h[0:7])
```

Comment:
I would like to see some unit tests for this. It is a little hard for a few reasons:
- result is a printed line
- interacts with git/ file system
- requires built binary

I think it could be testable if you put in a function and pass in binary flags. AKA:

`func warnGitHashMismatch(GitCommitHash) -> string`

by passing in the constant you can essentially mock the state of the binary. By returning the string you can test the output. The the layer above in the stack would be responsible for passing in the git hash (from build flag) and printing a string if returned. You could run tests like:
- In test get the current commit hash, pass it in and see that no warning is returned
- In test, get previous commit hash, pass it in and see warning is returned
- Pass in random hash (not in repo), and see no warning is returned (case of different repo)

Reply:
I added a new file `root_test.go` and reached 44.6% coverage - if needed, we could improve coverage further by adding test cases to the `rootCmd.SetVersionTemplate` code, though those parts may require a slight rewrite to make them more testable.   
Happy to pick it up separately after the PR is merged.

This is the first "production grade" go test file I write from scratch, please be thorough in your review if possible, and nit away! :)

Reply:
I am asking the group what to do here. Currently `make tests` fails because coverage is not over 80%, but it is an obvious improvement from the previous 0% so I would like to take as is. More on this...

Reply:
As discussed, with @nick-stroud I've disabled the new tests for the cmd package, and I'll take on a task to write tests to get coverage > 80%